x/attribute/keeper/keeper.go

```diff
@@ -104,7 +104,7 @@ func (k Keeper) SetAttribute(
 	ctx sdk.Context, acc sdk.AccAddress, attr types.Attribute, owner sdk.AccAddress,
 ) error {
 	// Ensure attribute is valid
-	if err := attr.Validate(); err != nil {
+	if err := attr.ValidateBasic(); err != nil {
```

Comment:
The comment removed below (re: stateful check on attribute length) -- this check should be added here in the keeper.